src/math/ChebyshevBoundedField.cc

```diff
@@ -427,15 +427,12 @@ PTR(BoundedField) ChebyshevBoundedField::operator*(double const scale) const {
 }
 
 bool ChebyshevBoundedField::operator==(BoundedField const& rhs) const {
-    try {
-        auto& rhsCasted = dynamic_cast<ChebyshevBoundedField const&>(rhs);
-        bool coefficientsEqual = _coefficients.getShape() == rhsCasted._coefficients.getShape() &&
-                                  all(equal(_coefficients, rhsCasted._coefficients));
-        return (getBBox() == rhsCasted.getBBox()) && coefficientsEqual;
-    }
-    catch (std::bad_cast&) {
-        return false;
-    }
+    auto rhsCasted = dynamic_cast<ChebyshevBoundedField const *>(&rhs);
+    if (!rhsCasted) return false;
```

Comment:
It's just avoiding (possible) overhead from catch/throw.  I'm not sure if the compiler should be able to optimize any of that away in this case, but I'm pretty sure some major ones wouldn't, at least as of a couple of years ago.